CONTRIBUTING.md: improve (#674)
ansible-collections/community.postgresql: CONTRIBUTING.md

# Contributing to this project

In this guide, you will find information relevant for code contributions, though any other kinds of contribution mentioned in the [Ansible Contributing guidelines](https://docs.ansible.com/ansible/devel/community/index.html) are equally appreciated and valuable.

> If you have any questions along the reading, please contact the community via one or more of the [available channels](https://github.com/ansible-collections/community.postgresql#communication). Any feedback on the guide is very welcome.

Reviewer: ```suggestion
If you have any questions after reading, please contact the community via one or more of the [available channels](https://github.com/ansible-collections/community.postgresql#communication). Any feedback on this guide is very welcome.
```

## Reviewing open issue and pull requests

Refer to the [review checklist](https://docs.ansible.com/ansible/devel/community/collection_contributors/collection_reviewing.html) when triaging issues or reviewing pull requests (hereinafter PRs).

Most important things to pay attention to:

> - Do not let major/breaking changes to sneak in a minor/bugfix release! All such changes should be discussed in a dedicated issue, added to a corresponding milestone (which can be found in Issues or created there), and merged right before the major release. Take a look at similar issues to see what needs to be done, reflect steps you did/need to do in the issue.

Reviewer: ```suggestion
- Do not let major/breaking changes sneak into a minor/bugfix release! All such changes should be discussed in a dedicated issue, added to a corresponding milestone (which can be found or created in the project's Issues), and merged right before the major release. Take a look at similar issues to see what needs to be done and reflect on the steps you did/need to do in the issue.
```

- Every PR (except doc, refactoring, test-related, or a PR containing a new module/plugin) contains a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). Let's give users a chance to know about the changes.

> - Every new module `DOCUMENTATION` section contains the `version_added: 'x.y.z'` field. Besides the informative purpose, it is used by the changelog-generating tool to add a corresponding entry to the changelog. As the project follows SemVer, it is typically a next minor (x.Y.z) version.

Reviewer: ```suggestion
- Every new module `DOCUMENTATION` section contains the `version_added: 'x.y.z'` field. Besides the informative purpose, it is used by the changelog-generating tool to add a corresponding entry to the changelog. As the project follows SemVer, it is typically a next minor (x.y.0) version.
```

Author: How about `x.Y.0` ? Many people don't know which one is minor. If you're OK, please edit the suggestion and commit it

> - Every new module argument contains the `version_added: 'x.y.z'` field. As the project follows SemVer, it is typically a next minor (x.Y.z) version.

Reviewer: x.y.z is not the next minor, x.y is. x.y.z is a 'point release' for bugfixes only, not new features (in semver)

Author: this is just to show the format, the explanation follows in the next sentence, isn't it enough? Feel free to suggest your wording

> - Every new module argument contains the `version_added: 'x.y.z'` field. As the project follows SemVer, it is typically a next minor (x.Y.z) version.

Reviewer: ```suggestion
- Every new module argument contains the `version_added: 'x.y.z'` field. As the project follows SemVer, it is typically a next minor (x.y.0) version.
```

Author: [same](https://github.com/ansible-collections/community.postgresql/pull/674#discussion_r1492220898) here

> - Non-refactoring code changes (bugfixes, new features) are covered with, at least, integration tests! There can be exceptions bug generally it is a requirement.

Reviewer: ```suggestion
- Non-refactoring code changes (bugfixes, new features) are covered with, at least, integration tests! There can be exceptions but generally it is a requirement.
```

## Code contributions

If you want to submit a bugfix or new feature, refer to the [Quick-start development guide](https://docs.ansible.com/ansible/devel/community/create_pr_quick_start.html) first.

> ## This project specifics

Reviewer: ```suggestion
## Project-specific info
```

We assume you have read the [Quick-start development guide](https://docs.ansible.com/ansible/devel/community/create_pr_quick_start.html).

> For any submitted PR in order to get merged, this project requires sanity, units, and integration tests to pass.

Reviewer: ```suggestion
In order for any submitted PR to get merged, this project requires sanity, unit, and integration tests to pass.
```

To keep the matrix relevant, we are subscribed to the [news-for-maintainers](https://github.com/ansible-collections/news-for-maintainers) repository and the [Collection maintainers & contributors](https://forum.ansible.com/g/CollectionMaintainer) forum group to track announcements affecting CI.

If our matrix is permanently outdated, for example, when supported `ansible-core` versions are missed, the collections can get excluded from the package, so keep it updated!

## Adding tests

If you are new here, read the [Quick-start development guide](https://docs.ansible.com/ansible/devel/community/create_pr_quick_start.html) first.

When fixing a bug, first reproduce it by adding a task as reported to a suitable file under the ``tests/integration/targets/<module_name>/tasks/`` directory and run the integration tests as described below. The same is relevant for new features.

It is not necessary but if you want you can also add unit tests to a suitable file under the ``tests/units/`` directory and run them as described below.

## Checking your code locally

> It will make your and other people's life a bit easier if you run the tests locally and fix all failures before pushing.

Reviewer: or at the very least, make your PR a 'draft' until said tests are all green?

Author: sounds very good to me, would you like to suggest the change?

> It will make your and other people's life a bit easier if you run the tests locally and fix all failures before pushing.

Reviewer: ```suggestion
It will make your and other people's life a bit easier if you run the tests locally and fix all failures before pushing. If you're unable to run the tests locally, please create your PR as a **draft** to avoid reviewers being added automatically.
```

Author: I do like this addition
